plugin/trino-hive/src/main/java/io/trino/plugin/hive/BackgroundHiveSplitLoader.java

```diff
@@ -137,7 +137,9 @@
             // Hive naming pattern per `org.apache.hadoop.hive.ql.exec.Utilities#getBucketIdFromFile()`
             Pattern.compile("(\\d+)_\\d+.*"),
             // Hive ACID with optional direct insert attempt id
-            BUCKET_WITH_OPTIONAL_ATTEMPT_ID_PATTERN);
+            BUCKET_WITH_OPTIONAL_ATTEMPT_ID_PATTERN,
+            // Spark-generated file, eg. part-00000-6666e260-fda8-4df5-a412-0a07a90b35a5_00000.c000.snappy.orc
+            Pattern.compile("part-(\\d{5})-[0-9a-f]{8}-[0-9a-f]{4}-[0-9a-f]{4}-[0-9a-f]{4}-[0-9a-f]{12}.*"));
```

Comment:
This sounds incorrect. We should not attempt to derived bucket number from a file path created by Spark, since the bucket number is meaningless to us. Why do you need this?

Reply:
It looks like the tables created by Spark in both native and Hive-compliant modes are marked as transactional, and Hive connector expects there to be a bucket number on all "original" files:
```                
// Hive requires "original" files of transactional tables to conform to the bucketed tables naming pattern, to match them with delete deltas.
int bucketId = getRequiredBucketNumber(originalFilePath);
```
Here's the stacktrace:
```
tests               | 	Caused by: java.lang.IllegalStateException: Cannot get bucket number from path: hdfs://hadoop-master:9000/user/hive/warehouse/test_presto_reading_spark_buckets/part-00000-cbacc54d-5b1d-4b57-a82f-1f0ca2728de0-c000.snappy.orc
tests               | 		at io.trino.plugin.hive.BackgroundHiveSplitLoader.lambda$getRequiredBucketNumber$11(BackgroundHiveSplitLoader.java:859)
tests               | 		at java.base/java.util.OptionalInt.orElseThrow(OptionalInt.java:271)
tests               | 		at io.trino.plugin.hive.BackgroundHiveSplitLoader.getRequiredBucketNumber(BackgroundHiveSplitLoader.java:859)
tests               | 		at io.trino.plugin.hive.BackgroundHiveSplitLoader.loadPartition(BackgroundHiveSplitLoader.java:533)
tests               | 		at io.trino.plugin.hive.BackgroundHiveSplitLoader.loadSplits(BackgroundHiveSplitLoader.java:337)
tests               | 		at io.trino.plugin.hive.BackgroundHiveSplitLoader$HiveSplitLoaderTask.process(BackgroundHiveSplitLoader.java:266)
tests               | 		... 6 more
```
The files created by Spark have a bucket number contained in the filename, but none of the patterns match them. Do you think I should rather look for a way to disable creation of transactional tables by Spark?

Reply:
Turns out disabling transactions is quite easy, we can go that route.

Reply:
> tables created by Spark in both native and Hive-compliant modes are marked as transactional


that's weird. Last time i checked, Spark did not support transactional tables natively.

Reply:
Maybe it's a feature of Spark 3? I found some comments about Spark not supporting transactions but it was for Spark 2.x.